src/harness/unittests/tsserverProjectSystem.ts

```diff
@@ -311,12 +312,21 @@ namespace ts.projectSystem {
         checkNumberOfInferredProjects(projectService, count.inferredProjects || 0);
     }
 
+    export function configuredProjectAt(projectService: server.ProjectService, index: number) {
+        const values = projectService.configuredProjects.values();
+        while (index > 0) {
+            values.next();
+            index--;
+        }
+        return values.next().value;
+    }
+
     export function checkWatchedFiles(host: TestServerHost, expectedFiles: string[]) {
         checkMapKeys("watchedFiles", host.watchedFiles, expectedFiles);
     }
 
-    export function checkWatchedDirectories(host: TestServerHost, expectedDirectories: string[]) {
-        checkMapKeys("watchedDirectories", host.watchedDirectories, expectedDirectories);
+    export function checkWatchedDirectories(host: TestServerHost, expectedDirectories: string[], recursive = false) {

```

Comment:
Good catch, i will add test case that need to check if the recursive is false (eg. files: [*.ts] in config file)